src/analysis/seminorm.lean

```diff
@@ -496,15 +578,51 @@ end normed_linear_ordered_field
 
 end seminorm
 
+/-! ### The norm as a seminorm -/
+
+section norm_seminorm
+variables (𝕜 E) [normed_field 𝕜] [semi_normed_group E] [semi_normed_space 𝕜 E] {r : ℝ}
+
+/-- The norm of a seminormed group as a seminorm. -/
+def norm_seminorm : seminorm 𝕜 E :=
+{ to_fun := norm,
+  smul' := norm_smul,
+  triangle' := norm_add_le }
+
```

Comment:
I feel like this should belong into `analysis/normed_space/basis`, but that would cause problems because that files contains the definition of `normed_field`, which we need for `seminorm`. But if that were not a problem, we could transfer quite a few lemmas from `normed_space` to `seminorm`.

Reply:
Also we have the following code to make a seminorm into a `normed_space`:
````
instance (s : seminorm 𝕜 E) : has_norm s.domain := ⟨s.to_fun⟩

lemma is_core (s : seminorm 𝕜 E) : semi_normed_group.core s.domain :=
  ⟨s.zero, s.triangle, s.neg⟩

instance (s : seminorm 𝕜 E) : semi_normed_group s.domain :=
  semi_normed_group.of_core s.domain s.is_core

lemma smul_le (s : seminorm 𝕜 E) (c : 𝕜) (x : E) : s (c • x) ≤ ∥c∥ * s x := by rw s.smul

instance (s : seminorm 𝕜 E) : semi_normed_space 𝕜 s.domain := {norm_smul_le := s.smul_le}
````

Reply:
There is a greater refactor to be done here, by making `seminorm` the primitive thing. The original problem is that @riccardobrasca didn't know about `seminorm` when first writing `semi_normed_group`.
But I think this is out of scope for this PR.

Reply:
I totally agree that sooner or later we have to do the refactor to have only one notion of `seminormed` stuff.